addon/components/au-button.js

```diff
@@ -1,39 +1,44 @@
 import Component from '@glimmer/component';
 
+const SKINS = ['primary', 'secondary', 'naked', 'link', 'link-secondary'];
+
 export default class AuButton extends Component {
-  get size() {
-    if (this.args.size == 'large') return 'au-c-button--large';
+  constructor() {
+    super(...arguments);
+
+    this.skin = SKINS.includes(this.args.skin)
+      ? this.args.skin == 'tertiary' // DEPRECATED
+        ? 'link'
+        : this.args.skin
+      : 'primary';
+  }
+
+  get sizeClass() {
+    if (this.args.size == 'large' && !this.skin.startsWith('link'))
+      return 'au-c-button--large';
     else return '';
   }
 
-  get width() {
+  get widthClass() {
     if (this.args.width == 'block') return 'au-c-button--block';
     else return '';
   }
 
-  get skin() {
-    if (this.args.skin == 'secondary') return 'au-c-button--secondary';
-    if (this.args.skin == 'naked') return 'au-c-button--naked';
-    if (this.args.skin == 'tertiary')
-      // DEPRECATED
-      return 'au-c-button--link';
-    if (this.args.skin == 'link') return 'au-c-button--link';
-    if (this.args.skin == 'link-secondary')
-      return 'au-c-button--link-secondary';
-    else return 'au-c-button--primary';
+  get skinClass() {
+    return `au-c-button--${this.skin}`;
```

Comment:
> I would keep the logic in a getter instead of setting `this.skin` in the constructor, that way it will rerun if the `@skin` argument value changes.

I forgot about that. I switched it to a getter 👍